src/NodeWrapper.ts

```diff
@@ -472,4 +474,86 @@ export default class NodeWrapper extends SelectableObject {
     this.nodeAcceptCircle.stroke(StateManager.colorScheme.nodeAcceptStrokeColor);
     this.nodeLabel.fill(StateManager.colorScheme.nodeLabelColor);
   }
+
+  public setErrorState(isError: boolean) {
+    this._isErrorState = isError;
+    this.updateAppearance();
+}
+
+
+private updateAppearance() {
+  // Remove existing error elements (if they exist) to avoid duplicates
+  this.nodeGroup.find('.errorIcon, .errorText').forEach((el) => el.destroy());
+
+  if (this._isErrorState) {
+      // Error state appearance
+      this.nodeBackground.fill('rgb(254, 226, 225)'); // Light red
+      this.nodeBackground.stroke('rgb(220, 38, 37)'); // Dark red
+      this.nodeBackground.strokeWidth(4);
+
+      // an "X" icon
+      const errorIcon = new Konva.Circle({
```

Comment:
Would it be possible to create these Konva objects on all nodes at the time of their creation, then just hide and show them depending on whether they have errors? I think that may be more performant than creating and destroying them every time their appearance needs to be updated.